pipelines.md

```diff
@@ -0,0 +1,323 @@
+-->
+
+## Creating new tasks and pipelines
+
+### The build, push and deploy pipeline
+
+- [build-deploy-pl.yaml](https://github.com/kabanero-io/kabanero-pipelines/blob/master/pipelines/incubator/build-deploy-pl.yaml)
+
+  This file is the primary pipeline that showcases all the tasks supplied in the Kabanero repo. It validates that the application stack is active, builds the application stack, publishes the application image to the container registry, does a security scan of the image, and conditionally deploys the application. When running the pipeline via a webhook, the pipeline leverages the triggers functionality to conditionally deploy the application only when a pull request is merged in the git repo.  Other actions that trigger the pipeline run, will validate, build, push, and scan the image.
+
+### Tasks
+
+- [build-task.yaml](https://github.com/kabanero-io/kabanero-pipelines/blob/master/pipelines/incubator/build-task.yaml)
+
+   This file builds a container image from the artifacts in the git-source repository by using [Buildah](https://github.com/containers/buildah). After the image is built, the `build-task` file publishes it to the docker-image URL by using Buildah.
+
+- [deploy-task.yaml](https://github.com/kabanero-io/kabanero-pipelines/blob/master/pipelines/incubator/deploy-task.yaml)
+
+   This file modifies the `app-deploy.yaml` file, which describes the deployment options for the application. `Deploy-task` modifies `app-deploy.yaml` to point to the image that was published and deploys the application by using the application deployment operator. Generate your `app-deploy.yaml` file by running the `appsody deploy --generate-only` command.
+
+By default, the pipelines run and deploy the application in the `kabanero` namespace. If you want to deploy the application in a different namespace, update the `app-deploy.yaml` file to point to that namespace.
+
+For more information, see [the kabanero-pipelines repo](https://github.com/kabanero-io/kabanero-pipelines).
+
+<!--
+// =================================================================================================
+// Running pipelines
+// =================================================================================================
+-->
+
+## Running pipelines
+
+Explore how to use pipelines to build and manage application stacks.
+
+### Prerequisites
+
+1. [Kabanero foundation](https://github.com/kabanero-io/kabanero-foundation) must be installed on a supported Kubernetes deployment.
+
+1. [A pipelines dashboard](https://github.com/tektoncd/dashboard) is installed by default with Kabanero's Kubernetes operator. To find the pipelines dashboard URL, login to your cluster and run the `oc get routes` command or in the Kabanero landing page.
+
+1. A persistent volume must be configured. See the following section for details.
+
+1. Secrets for the git repo (if private) and image repository
+
+### Getting started
+
+#### Setting up a persistent volume to run pipelines
```

Comment:
Looks like an h4 does not do well with the markdown CSS and becomes light grey. The h5 title that follows is actually larger and black, so the sequencing here looks all wrong. Rather than presenting @alohr51 with a problem very late in the day (should still be addressed so will raise an issue), I suggest we reformat this section. 

It looks to me that "Getting started" consists of 2 steps: Setting up a persistent volume and Creating secrets. Let's begin the section with a lead in sentence:

Follow these steps:

Change  #### Setting up a persistent volume to run pipelines to an ordered list (1) and #### Creating secrets to (2).

Static persistent volumes and Dynamic volume provisioning could become sublist items. 

I tested this locally and it seems to look ok.

Reply:
Okay